src/seasidecontactbuilder.cpp

```diff
@@ -0,0 +1,666 @@
+    : d(new SeasideContactBuilderPrivate)
+{
+    // defaults.  override in the ctor of your derived type.
+    d->manager = 0;
+    d->propertyHandler = 0;
+    d->unimportableDetailTypes = (QSet<QContactDetail::DetailType>() << QContactDetail::TypeGlobalPresence << QContactDetail::TypeVersion);
+    d->importableSyncTargets = (QStringList() << QLatin1String("was_local") << QLatin1String("bluetooth"));
+}
+
+SeasideContactBuilder::~SeasideContactBuilder()
+{
+    delete d->propertyHandler;
+    delete d;
+}
+
+/*
+ * Returns a pointer to a valid QContactManager.
+ * The default implementation uses the SeasideCache manager.
+ */
+QContactManager *SeasideContactBuilder::manager()
+{
+    if (!d->manager) {
+        d->manager = SeasideCache::manager();
+    }
+
+    return d->manager;
+}
+
+/*
+ * Returns a filter which will return the subset of contacts
+ * in the manager which are potential merge candidates for
+ * the imported contacts (ie, come from the same sync target).
+ *
+ * The default implementation will return a filter which matches
+ * any local / was_local / Bluetooth contact.
+ */
+QContactFilter SeasideContactBuilder::mergeSubsetFilter() const
+{
+    return localContactFilter();
+}
+
+/*
+ * Returns a versit property handler which will be used during
+ * conversion of versit documents into QContact instances.
+ *
+ * The default implementation will return a SeasidePropertyHandler.
+ */
+QVersitContactHandler *SeasideContactBuilder::propertyHandler()
+{
+    if (!d->propertyHandler) {
+        d->propertyHandler = new SeasidePropertyHandler;
+    }
+
+    return d->propertyHandler;
+}
+
+/*
+ * Merge the given (matching) \a local contact into the given
+ * \a import contact, so that the \a import contact could be
+ * later saved in the manager.
+ *
+ * Returns \c true if the \a import contact differed significantly
+ * from the \a local contact (that is, if saving the returned
+ * \a import contact would result in the \a local contact being
+ * updated).
+ *
+ * The \a erase value will be set to true if the given import
+ * contact should be erased from the imported contacts list if
+ * there are no significant differences between it and the local
+ * contact (that is, if the imported contact should be omitted
+ * from later possible store operations, due to the fact that such
+ * a store operation would be a no-op for that contact).
+ *
+ * The default implementation performs a non-destructive merge
+ * and will set \a erase to true, which is the required behaviour
+ * for an "import" style sync.
+ *
+ * Sync implementations which require remote-detail-removal
+ * semantics, for example, should implement this function
+ * differently (eg, prefer-local or prefer-remote), and should
+ * possibly set \a erase to false if they don't wish to prune
+ * the import list prior to save.
+ */
+bool SeasideContactBuilder::mergeImportWithLocal(QContact &import, const QContact &local, bool *erase)
```

Comment:
It would be better to use the "merge into" form than "merge with".  Also applies to 'mergeImportWithImport'.